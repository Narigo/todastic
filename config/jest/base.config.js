const path = require("path");

module.exports = {
  coverageDirectory: "coverage",
  rootDir: path.resolve(__dirname, "../../"),
  moduleFileExtensions: ["ts", "tsx", "js"],
  testMatch: ["<rootDir>/src/**/?(*.)spec.ts"],
  testPathIgnorePatterns: ["<rootDir>/node_modules/"],
  transform: {
    "^.+\\.ts$": "<rootDir>/node_modules/babel-jest"
  },
  verbose: true
};
