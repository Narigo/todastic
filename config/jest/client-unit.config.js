const config = require("./unit.config.js");

module.exports = {
  ...config,
  roots: ["<rootDir>/src/client/lib", "<rootDir>/src/common/lib"]
};
