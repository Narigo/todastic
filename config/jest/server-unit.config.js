const config = require("./unit.config.js");

module.exports = {
  ...config,
  roots: ["<rootDir>/src/server/lib", "<rootDir>/src/common/lib"],
  coverageThreshold: {
    "**/service/**/*.js": {
      branches: 0,
      functions: 0,
      lines: 0,
      statements: 0
    }
  }
};
