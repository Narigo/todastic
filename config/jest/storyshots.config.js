const config = require('./base.config.js');

module.exports = {
  ...config,
  testEnvironment: 'jest-environment-jsdom',
  testMatch: ['<rootDir>/config/jest/storyshots.spec.ts']
};
