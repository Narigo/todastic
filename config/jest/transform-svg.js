const path = require('path');

// This is a custom Jest transformer turning svg imports into empty objects.
// http://facebook.github.io/jest/docs/en/webpack.html

module.exports = {
  process(src, filename) {
    const viewBoxMatch = /viewBox="([^"]*?)"/.exec(src);
    const viewBox = viewBoxMatch ? viewBoxMatch[1] : '';
    return `module.exports = {
      id: ${JSON.stringify(path.basename(filename))},
      viewBox: '${viewBox}'
    };`;
  }
};
