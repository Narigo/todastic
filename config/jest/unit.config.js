const config = require("./base.config.js");

module.exports = {
  ...config,
  collectCoverage: true,
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100
    }
  },
};
