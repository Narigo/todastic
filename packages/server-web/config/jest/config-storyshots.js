const config = require('./config');

module.exports = {
  ...config,
  testEnvironment: 'jest-environment-jsdom',
  testMatch: ['<rootDir>/config/jest/storyshots.spec.js']
};
