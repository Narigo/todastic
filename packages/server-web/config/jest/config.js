module.exports = {
  rootDir: "../../",
  testMatch: ["<rootDir>/src/**/?(*.)spec.js"],
  transform: {
    "^.+\\.svg$": "<rootDir>/config/jest/svg-transform.js",
    "^(?!.*\\.(vue|js|json|svg)$)": "<rootDir>/config/jest/file-transform.js",
    "^.+\\.js$": "babel-jest",
    "^.+\\.vue$": "vue-jest"
  },
  collectCoverage: true,
  collectCoverageFrom: [
    "<rootDir>/src/**/*.{js}",
    "!**/node_modules/**",
    "!**/*.config.js",
    "!**/client/**",
    "!**/service/**",
    "!**/dist/**"
  ],
  coverageReporters: ["json"],
  coverageDirectory: "<rootDir>/../../coverage/server-web"
};
