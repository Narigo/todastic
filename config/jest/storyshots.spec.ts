import util from "util";
import initStoryshots from "@storybook/addon-storyshots";
import registerRequireContext from "babel-plugin-require-context-hook/register";
import "jest";

// Storybook uses require.context() - we need to replace it for Jest
registerRequireContext();

// We need to catch proptype warnings in the tests.

// Keep a reference to the original console methods.
/* eslint-disable no-console */
const consoleWarn = console.warn;
const consoleError = console.error;

function logToError(format, ...args) {
  throw new Error(util.format(format, ...args));
}

beforeEach(() => {
  console.warn = logToError;
  console.error = logToError;
});

afterEach(() => {
  console.warn = consoleWarn;
  console.error = consoleError;
});
/* eslint-enable no-console */

initStoryshots({
  configPath: "config/storybook/config.ts"
});
