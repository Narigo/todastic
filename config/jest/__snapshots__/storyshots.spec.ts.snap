// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Button default 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Some text on button
</button>
`;

exports[`Storyshots Components/Button disabled 1`] = `
<button
  className="undefined"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  Disabled
</button>
`;

exports[`Storyshots Components/DeleteMe default 1`] = `
<div
  className=""
>
  <div>
    DeleteMe
  </div>
</div>
`;

exports[`Storyshots Components/DeleteMe modifier 1`] = `
<div
  className="undefined"
>
  <div>
    DeleteMe
  </div>
</div>
`;

exports[`Storyshots Components/LoginForm default 1`] = `
<div
  className=""
>
  <form
    onSubmit={[Function]}
  >
    <div>
      Login
    </div>
    <div>
      <label>
        <span>
          Username
        </span>
        <input
          autoComplete="username"
          className=""
          disabled={false}
          onChange={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label>
        <span>
          Passwort
        </span>
        <input
          autoComplete="current-password"
          className=""
          disabled={false}
          onChange={[Function]}
          placeholder=""
          type="password"
          value=""
        />
      </label>
    </div>
    <div>
      <button
        className=""
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Login
      </button>
    </div>
  </form>
  <div />
</div>
`;

exports[`Storyshots Components/LoginForm isError 1`] = `
<div
  className="undefined"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      Login
    </div>
    <div>
      <label>
        <span>
          Username
        </span>
        <input
          autoComplete="username"
          className=""
          disabled={false}
          onChange={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label>
        <span>
          Passwort
        </span>
        <input
          autoComplete="current-password"
          className=""
          disabled={false}
          onChange={[Function]}
          placeholder=""
          type="password"
          value=""
        />
      </label>
    </div>
    <div>
      <button
        className=""
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Login
      </button>
    </div>
  </form>
  <div>
    This is an error!
  </div>
</div>
`;

exports[`Storyshots Components/LoginForm isWarning 1`] = `
<div
  className="undefined"
>
  <form
    onSubmit={[Function]}
  >
    <div>
      Login
    </div>
    <div>
      <label>
        <span>
          Username
        </span>
        <input
          autoComplete="username"
          className=""
          disabled={false}
          onChange={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label>
        <span>
          Passwort
        </span>
        <input
          autoComplete="current-password"
          className=""
          disabled={false}
          onChange={[Function]}
          placeholder=""
          type="password"
          value=""
        />
      </label>
    </div>
    <div>
      <button
        className=""
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Login
      </button>
    </div>
  </form>
  <div>
    This is a warning.
  </div>
</div>
`;

exports[`Storyshots Components/TextInput default 1`] = `
<input
  autoComplete="off"
  className=""
  disabled={false}
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;

exports[`Storyshots Components/TextInput disabled text 1`] = `
<input
  autoComplete="off"
  className="undefined"
  disabled={true}
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;

exports[`Storyshots Components/TextInput placeholder 1`] = `
<input
  autoComplete="off"
  className=""
  disabled={false}
  onChange={[Function]}
  placeholder="hello placeholder"
  type="text"
  value=""
/>
`;

exports[`Storyshots Components/TodoLabel default 1`] = `
<span
  className=""
>
  #
  test
</span>
`;

exports[`Storyshots Components/TodoLabel selected 1`] = `
<span
  className="undefined"
>
  #
  some-label
</span>
`;

exports[`Storyshots Components/TodoLabel track call 1`] = `
<span
  className=""
>
  #
  track
</span>
`;

exports[`Storyshots Components/TodoStatus default 1`] = `
<div
  className=""
  onClick={[Function]}
/>
`;

exports[`Storyshots Components/TodoStatus done 1`] = `
<div
  className="undefined"
  onClick={[Function]}
/>
`;
